datafusion/optimizer/src/eliminate_cross_join.rs

```diff
@@ -176,159 +178,138 @@ fn flatten_join_inputs(
 fn find_inner_join(
     left_input: &LogicalPlan,
     rights: &mut Vec<LogicalPlan>,
-    possible_join_keys: &mut Vec<(Expr, Expr)>,
-    all_join_keys: &mut HashSet<(Expr, Expr)>,
+    possible_join_predicates: &mut Vec<Expr>,
 ) -> Result<LogicalPlan> {
+    let mut candidate_right_input: Option<(usize, Vec<Expr>)> = None;
     for (i, right_input) in rights.iter().enumerate() {
-        let mut join_keys = vec![];
-
-        for (l, r) in &mut *possible_join_keys {
-            let key_pair = find_valid_equijoin_key_pair(
-                l,
-                r,
-                left_input.schema().clone(),
-                right_input.schema().clone(),
-            )?;
+        let predicate_schemas =
```

Comment:
we should find column compare pair. like `t1.a > t2.a` or `t1.a > t2.a + 1`.
If left or right don't contains column, we shouldn't put them into join_filter.